packages/renderer/src/lib/dialogs/CustomPick.svelte

```diff
@@ -0,0 +1,200 @@
+$: itemSectionHiddenStatus = new Map<number, boolean>();
+
+onMount(() => {
+  // handle the showCustomPick events
+  window.events?.receive('showCustomPick:add', showCustomPickCallback);
+});
+
+async function showCustomPickCallback(options?: CustomPickOptions) {
+  id = options.id;
+  title = options.title;
+  description = options.description;
+  icon = options.icon;
+  items = options.items;
+  canSelectMany = options.canSelectMany;
+  hideItemSections = options.hideItemSections;
+
+  items.forEach((_value, index) => {
+    itemSectionHiddenStatus.set(index, hideItemSections);
+  });
+  display = true;
+}
+
+function handleSelection(
+  e: MouseEvent & {
+    currentTarget: EventTarget & HTMLDivElement;
+  },
+  item: CustomPickItem,
+) {
+  if (
+    e.target instanceof HTMLButtonElement &&
+    (e.target.ariaLabel === 'Less detail' || e.target.ariaLabel === 'Show more')
+  ) {
+    return;
+  }
+  //if it's deselecting an item, just do it
+  //if it's selecting an item and the multiselect is disabled, first it unselects any selected item before selecting it
+  if (!item.selected && !canSelectMany) {
+    items.forEach(item => {
+      item.selected = false;
+    });
+  }
+  item.selected = !item.selected;
+  items = items;
+}
+
+async function cancel() {
+  await window.closeCustomPick(id);
+  display = false;
+}
+
+async function next() {
+  const indexes = [];
+  for (let i = 0; i < items.length; i++) {
+    if (items[i].selected) {
+      indexes.push(i);
+    }
+  }
+  await window.sendCustomPickItemsOnConfirmation(id, indexes);
+  await window.closeCustomPick(id);
+  display = false;
+}
+
+function setSectionVisibility(index: number, show: boolean) {
+  itemSectionHiddenStatus.set(index, show);
+  itemSectionHiddenStatus = itemSectionHiddenStatus;
+}
+</script>
+
+{#if display}
+  <!-- Create overlay-->
+  <div
+    class="fixed top-0 left-0 right-0 bottom-0 bg-black pt-8 pb-3 bg-opacity-60 bg-blend-multiply h-full grid z-50 overflow-y-auto">
+    <div class="flex flex-col place-self-center w-[650px] rounded-xl bg-charcoal-800 shadow-xl shadow-black">
+      <div class="flex flex-col items-center justify-between pl-4 pr-3 py-3 space-x-2 text-gray-400">
+        {#if icon}
+          <div class="mb-2">
+            {#if typeof icon === 'string'}
+              <img src="{icon}" alt="{title ? title : 'icon'}" class="max-h-12" />
+            {:else}
+              <!-- TODO check theme used for image, now use dark by default -->
+              <img src="{icon.dark}" alt="{title ? title : 'icon'}" class="max-h-12" />
+            {/if}
+          </div>
+        {/if}
+        {#if title}
+          <h1 class="grow text-lg font-bold capitalize" aria-label="{title}">{title}</h1>
+        {/if}
+        {#if description}
+          <h3 class="grow text-xs">{description}</h3>
+        {/if}
+      </div>
+
+      <div class="flex flex-row flex-wrap mx-auto">
+        {#each items as item, i}
```

Comment:
FYI we're not doing it yet in our components, so mainly a FYI but we should add keys
https://svelte.dev/tutorial/keyed-each-blocks

Reply:
by looking at the example it uses the property id of the item as a key. In my case i don't have anything like that. I suppose i should use the array index? `{#each items as item, i (i)}`